e2e/browser/src/components/notifications/index.tsx

```diff
@@ -21,13 +21,127 @@ const MessageList = (props: { messages: Array<any> }) => {
   );
 };
 
+const WebSocketButtons = ({ socket }: { socket?: WebsocketNotification }) => {
```

Comment:
better to pass the handlers in here, rather than the socket itself, as the socket is a pretty complex object which may cause state changes.

Reply:
1ffd2f6581286fd5ef9541ef04f6a5343a77b93d